src/components/Search/FreeSearchForm.tsx

```diff
@@ -1,132 +1,135 @@
+
   const [searchState, setSearchState] = useState<ISearchState>({
-    selections: [],
     searchWord: '',
     resultObject: {},
   });
 
-  useEffect(() => {
-    setSearchState((current) => ({
-      ...current,
-      selections: Object.keys(freeSearchParams || {}),
-    }));
-  }, [freeSearchParams]);
-
-  const { selections, searchWord, resultObject } = searchState;
+  const { searchWord, resultObject } = searchState;
 
   const getSuggestions = useCallback(
     (inputValue: string) =>
       new Promise<{ value: string; label: string }[]>((resolve, reject) => {
         getProjectsWithFreeSearch(inputValue)
           .then((res) => {
-            if (res) {
-              // Create a combined list of all results and filter already selected values
-              const resultList = freeSearchResultToList(res).filter(
-                (f) => !arrayHasValue(selections, f.value),
-              );
-
-              // Convert the resultList to options for the suggestion dropdown
-              const freeSearchFormItemList: Array<FreeSearchFormItem> | [] = resultList
-                ? resultList.map((r) => ({ ...listItemToOption(r), type: r.type }))
-                : [];
-
-              // This resultObject is needed to be able to add the type, value and name of the selected option,
-              // since the callback for selecting an option only returns the displayed value instead of the mapped object.
-              if (freeSearchFormItemList.length > 0) {
-                setSearchState((current) => ({
-                  ...current,
-                  resultObject: _.keyBy(freeSearchFormItemList, 'label'),
-                }));
-              }
-
-              resolve(freeSearchFormItemList);
+            const formValue = getValues('freeSearchParams');
+            // Create a combined list of all results and filter already added values
+            const resultList = freeSearchResultsToList(res || {}).filter(
+              (f) => !arrayHasValue(Object.keys(formValue), f.value),
+            );
+
+            // Convert the resultList to options for the suggestion dropdown
+            const freeSearchFormItemList: Array<FreeSearchFormItem> | [] = resultList
+              ? resultList.map((r) => ({ ...listItemToOption(r), type: r.type }))
+              : [];
+
+            // This resultObject is needed to be able to add the type, value and name of the selected option,
+            // since the callback for selecting an option only returns the displayed value instead of the mapped object.
+            if (freeSearchFormItemList.length > 0) {
+              setSearchState((current) => ({
+                ...current,
+                resultObject: _.keyBy(freeSearchFormItemList, 'label'),
+              }));
             }
-            resolve([]);
+            resolve(freeSearchFormItemList || []);
           })
           .catch(() => reject([]));
       }),
-    [selections],
+    [],
```

Comment:
add getValues as dependency, because it is coming from outside the component

Reply:
fixed